Add section on siunitx (#33)
oetiker/lshort: book/src/math.tex

  \IEEEyessubnumber  
\end{IEEEeqnarray}
\end{example}

\section{Units} \label{sec:units} \index{units (SI)}\index{SI}

When dealing with real world data you will often find yourself writing units
such as \qty{10}{\kg} or \qty{25}{\coulomb\per\mole}. In technical writing it

> is crucial to accurately convey the values so errors and ambiguity are avoided.

Reviewer: ```suggestion
is crucial to accurately convey values so that errors and ambiguity are avoided.
```

> Therefore a system of units known as \emph{International System of Units}, also
> known simply as \emph{SI}, was defined \cite{si}. Along with it a detailed set
> of typesetting rules is defined. For example you may have noticed that in the

Reviewer: ```suggestion
The \emph{International System of Units} (\emph{SI}) \cite{si}, comes with a detailed set
of typesetting rules. For example you may have noticed that in the
```

first sentence the space between \num{10} and \unit{\kg} is smaller than
between the words.

Fortunately thanks to the logical nature of \LaTeX{} markup and the excellent

> \pai*{siunitx} package you do not need to know most of the rules. If you
> consequently use \pai{siunitx} commands then units in your document will be
> typeset correctly.

Reviewer: ```suggestion
\pai*{siunitx} package you do not need to know most of the rules. Just use the special unit-commands provided by the package and all the units in your document will be
typeset correctly.
```

\subsection{Pitfalls of naively entered units}

> When initially writing units you may have simply written

Reviewer: ```suggestion
Naïvely, one might write units like this:
```

\begin{example}[examplewidth=6cm]
The speed of light
is 299792458 m/s.
\end{example}

This may look fine initially but if your document gets longer you will probably
get into situation where unfortunate linebreak will occur 
\begin{example}[examplewidth=6cm]
\ldots{} The speed of light
is 299792458 m/s. Thus we
can calculate \ldots
\end{example}

> Obviously we can use non-breaking space in order to fix it, but this is not the
> end of the problems. Another thing is spacing between the numbers. Digits of

Reviewer: ```suggestion
A non-breaking space will fix this problem, but the next one is already waiting---the space between numbers. Digits of
```

long numbers should be grouped to make them easier to read. The above example
should then written like this.
\begin{example}[examplewidth=6cm]
The speed of light is
299\,792\,458~m/s.
\end{example}

> Yet another problem is encountered when trying to write the units inside math

Reviewer: ```suggestion
Yet another problem arrises when trying to write the units inside math
```

mode
\begin{example}[examplewidth=6cm]
$10~\frac{m}{s} \cdot 12~s = 120~m$
\end{example}

> Here we also need to make sure that the units are written in roman font.

Reviewer: ```suggestion
Here we need to make sure that the units are written in roman font.
```

The above example should be then written as
\begin{example}[examplewidth=6cm]
$10~\frac{\mathrm{m}}{\mathrm{s}} 
  \cdot 12~\mathrm{s} 
  = 120~\mathrm{m}$
\end{example}
Hopefully these examples illustrate why a better method is needed.

\subsection{Basic commands of the \pai{siunitx} package}

The basic commands of \pai{siunitx} package are
\begin{lscommand}
  \ci{num}[options][number]\\
  \ci{unit}[options][unit]\\
  \ci*{qty}[options:o ! number:m ! unit:m] 
\end{lscommand}
The \ci{num} command typesets a number, \ci{unit} typesets a unit, and \ci{qty}
typesets a quantity, i.e.\ a number followed by a unit. All of the commands

> provide optional \carg{options} argument, which is a comma delimited key value

Reviewer: ```suggestion
provide optional \carg{options} argument, which is a comma delimited key-value pair
```

recognise the logical structure of the units, which means that the formatting
can be changed on the fly.
\begin{example}[examplewidth=0.32\linewidth]
Which one do you prefer? \\
$\unit{\newton} = \unit{
  \kilo\gram\metre\per\square\second}$ \\
$\unit{\newton} = \unit[per-mode = fraction]{
  \kilo\gram\metre\per\second\squared}$ \\
$\unit{\N} = \unit[per-mode = symbol]{
  \kg\m\per\s\tothe{2}}$
\end{example}
The commands \ci{gram}, \ci{kilo}, \ci{per} are rather self explanatory. As you
can see we have also used the \cargv{per-mode} option to influence the unit
style. While the command \ci{tothe} typesets arbitrary superscripts, the
command \ci{of} typesets arbitrary subscripts, known as qualifiers.
\begin{example}[examplewidth=0.32\linewidth]
\unit{\kg\of{mol}}, \unit{\bel\of{i}}
\end{example}

Note that the \ci{per} command only works on the next unit. If we want it to
apply to all following units we may use \cargv{sticky-per} option.
\begin{example}[examplewidth=0.4\linewidth]
The unit of thermal conductivity is
\unit{\watt\per\metre\per\kelvin}
or equivalently
\unit[sticky-per]{\m\kg\per\s\cubed\K}. 
\end{example}

> Sometimes you may want to make the units use a different colour so they stick

Reviewer: ```suggestion
Sometimes you may want to make the units use a different colour so they figure more prominently
```

> Sometimes you may want to make the units use a different colour so they stick
> out more. This can be done using \cargv{unit-color}. Options

Reviewer: ```suggestion
Sometimes you may want to make the units use a different colour so they figure more prominenty. This can be done using \cargv{unit-color}. Options
```

also be done in several ways. In order to customise the appearance of
uncertainty use the \cargv{uncertainty-mode} option.
\begin{example}[examplewidth=0.4\linewidth]
$M_\oplus = \qty{
  5.9722(6) e24}{\kg}$ \\
$M_\oplus = \qty{
  5.9722 +- 0.0006 e24}{\kg}$ \\
$M_\oplus = \qty{
  5.9722 \pm 0.0006 e24}{\kg}$ \\
$M_\oplus = \qty[
    uncertainty-mode = separate
  ]{5.9722(6) e24}{\kg}$ \\
$M_\oplus = \qty[
  uncertainty-mode = full
]{5.9722(6) e24}{\kg}$
\end{example}

By default \pai{siunitx} typesets the quantities in math mode. This may be
especially jarring if you use different font for the document text. One simple
solution is to set the \cargv{mode} option to \cargv{text}.
% TODO: Add pointer to changing math fonts section when written.
\begin{example}[examplewidth=0.4\linewidth]
\setmainfont{Liberation Sans}

\qty{1}{\degreeCelsius} is
equal to \qty{1}{\kelvin}, but
\qty[mode = text]{0}{\degreeCelsius}
is \qty[mode = text]{-273.15}{\kelvin}.
Bizarre, isn't it?
\end{example}

The behaviour of the commands may be customised even further. For a full list
of options check out the \pai{siunitx} package documentation.

> To avoid repeating the options with every use of commands, use

Reviewer: ```suggestion
To avoid repeating the options with every use of commands, use the
```

\begin{lscommand}
\ci{sisetup}[options]
\end{lscommand}
command. It can be used in the preamble to ensure consistent style throughout
the document, or inside the document when the style needs to be changed just

> in a given element, such as table. These options may be also passed as package

Reviewer: ```suggestion
in a given element, such as a table. These options may be also passed as package
```

quantities.
\begin{example}
\DeclareSIUnit{\pt}{pt}
The default font size
in \LaTeX{} is \qty{10}{\pt}.
\end{example}

This command may also be used when we want to adjust the appearance of
predefined unit. For example, the default litre is typeset using uppercase
\enquote*{L} and you may want to change it to use lowercase \enquote*{l}
instead.
\begin{example}[examplewidth=0.4\linewidth]
Litre before redefining: \unit{\L}. \\
\DeclareSIUnit{\litre}{l}
Litre after redefining: \unit{\L}.
\end{example}

Similarly the commands \ci{DeclareSIPrefix}, \ci{DeclareSIPower} and
\ci{DeclareSIQualifier} allow definition of macros in case additional are
needed.
\begin{example}[examplewidth=0.4\linewidth]
\DeclareSIUnit{\pt}{pt}
\DeclareSIPower\quartic\tothefourth{4}
\DeclareSIPrefix\decakilo{dk}{4}
\DeclareSIQualifier\polymer{pol}

It's over \qty{9000}{\quartic\decakilo\pt\polymer}!
\end{example}

\subsection{Table columns with numbers} \label{sec:sitables} \index{decimal alignment}

When typesetting tables which contain numeric data it is often useful to align
them along the decimal point, so they can be compared easily. The \pai{siunitx}

> package adds the special column specifier \cargv{S} for the \ei{tabular}
> environment, that can be used for such purpose. Non numeric data must be

Reviewer: ```suggestion
package adds the special column specifier \cargv{S} for this purpose to the \ei{tabular}
environment. Non numeric data must be
```
